lib/SocketHelpers.cpp

```diff
@@ -102,6 +102,21 @@ static bool ReceiveBytes(const wil::unique_socket& socket, gsl::span<uint8_t> bu
 {
     while (buffer.size_bytes() > 0)
     {
+        // Wait until there is something to read (or fail after a timeout)
+        const timeval timeout{0, 500000}; // 0.5 sec timeout
```

Comment:
Suggest to use designated initializers here to make the timeout value a bit more obvious.

```suggestion
        const timeval timeout{.tv_sec = 0, .tv_usec = 500000}; // 0.5 sec timeout
```

Reply:
Right, now that we are in C++20, we can do that too!